src/sqs/mod.rs

```diff
@@ -0,0 +1,271 @@
+    pub fn from_config(config: &Config, logger: &RBLogger) -> SyncEventQueue {
+        let region = env::var("AWS_LOCAL_SQS")
+            .map(|endpoint| Region::Custom {
+                endpoint,
+                name: "env_var".to_string(),
+            })
+            .unwrap_or_default();
+        // "https://sqs.us-east-1.amazonaws.com/927034868273/fxa-oauth-account-change-dev"
+        let queue_url = config
+            .get_str("sqs_url")
+            .unwrap_or("https://sqs.us-east-1.amazonaws.com/927034868273/rust_sqs_test");
+        // Max wait_time is 20, so we'll pound away on SQS.
+        SyncEventQueue {
+            sqs: Rc::new(SqsClient::simple(region)),
+            url: queue_url.to_owned(),
+            logger: logger.clone(),
+        }
+    }
+
+    //to "ack" an SQS message, you delete it from the queue.
+    pub fn ack_message(&self, event: &SyncEvent) -> Result<()> {
+        let sqs = self.sqs.clone();
+        let mut request = DeleteMessageRequest::default();
+        request.queue_url = self.url.clone();
+        request.receipt_handle = event.handle.clone();
+        match sqs
+            .delete_message(&request)
+            .with_timeout(Duration::new(1, 0))
+            .sync()
+        {
+            Ok(_) => Ok(()),
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    // Fetch message from sqs
+    pub fn fetch(&self) -> Option<SyncEvent> {
+        let sqs = self.sqs.clone();
+        // capture response via retry_if(move||{..})
+        let mut request = ReceiveMessageRequest::default();
+        request.queue_url = self.url.clone();
+        match sqs
+            .receive_message(&request)
+            .with_timeout(Duration::new(1, 0))
+            .sync()
+        {
+            Ok(r) => {
+                slog_debug!(self.logger.log, "Got message; {:?}", r);
+                if let Some(m) = r.messages {
+                    slog_debug!(self.logger.log, "Messages {:?}", m);
+                    // Ack the message
+                    let event: SyncEvent = m[0].clone().into();
+                    slog_debug!(self.logger.log, "Event: {:?}", event);
+                    if event.event.to_lowercase() == "delete"
+                        || event.event.to_lowercase() == "device:delete"
+                    {
+                        return Some(event);
+                    }
+                }
+                None
+            }
+            Err(e) => {
+                slog_warn!(self.logger.log, "Error: {:?}", e);
+                None
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    use rand;
+    use std::env;
+    use std::rc::Rc;
+    use std::thread;
+    use std::time::Duration;
+
+    use rocket::config::{self, Config, Table, Value};
+    use rusoto_core::Region;
+    use rusoto_sqs::{self, Sqs, SqsClient};
+
+    use super::SyncEventQueue;
+    use error::Result;
+    use logging::RBLogger;
+
+    fn setup(queue_url: String) -> SyncEventQueue {
```

Comment:
how do these tests run locally (what's the AWS_LOCAL_SQS we talk to?)

Reply:
They don't run locally. AWS_LOCAL_SQS points to the AWS local region that is used for SQS. To test against SQS, you'll need to create a test SQS server, preload it with messages and use the configuration `sqs_url`. I've been using "https://sqs.us-east-1.amazonaws.com/927034868273/rust_sqs_test" for personal testing.